src/components/product.js

```diff
@@ -619,20 +601,21 @@ export default class Product extends Component {
     evt.preventDefault();
     const selectedImageId = target.getAttribute('data-image-id');
     const imageList = this.model.images;
-    const foundImage = imageList.filter((image) => {
-      return image.id === parseInt(selectedImageId, 10);
-    })[0];
+    const foundImage = imageList.find((image) => {
+      return image.id === selectedImageId;
+    });
 
     if (foundImage) {
       this.selectedImage = foundImage;
       this.cachedImage = foundImage;
     }
+
     this.view.render();
   }
 
   nextIndex(currentIndex, offset) {
     const nextIndex = currentIndex + offset;
-    if (nextIndex > this.model.images.length) {
+    if (nextIndex > this.model.images.length - 1) {
```

Comment:
couldn't this be more easily written as `if (nextIndex === this.model.images.length)` and skip the math? Or if you're paranoid, use `>=`?